back/src/vhu/where.ts

```diff
@@ -0,0 +1,55 @@
+import { Prisma } from "@prisma/client";
+import { UserInputError } from "apollo-server-express";
+import { safeInput } from "../forms/form-converter";
+import { BsvhuWhere, DateFilter } from "../generated/graphql/types";
+
+export function convertWhereToDbFilter(
+  where: BsvhuWhere
+): Prisma.BsvhuFormWhereInput {
+  if (!where) {
+    return {};
+  }
+
+  const { _or, _and, _not, ...filters } = where;
+
+  const hasOrNesting = _or?.some(w => w._or || w._and || w._not);
```

Comment:
si on a un nesting en sautant un niveau {_or: [ {lorem:  { _and: […]}, {…}] } il me semble que ça laisse passer

Reply:
Normalement [le modèle](https://github.com/MTES-MCT/trackdechets/pull/748/files#diff-1ac998cd1ad8d39ace4f201f815cdee0fe0e3604bad495a4241f7a4e7faaf6c8R115) que j'ai mis en place ne le permet pas: on peut mettre un `BsvhuWhere` uniquement dans un `or / and / not` (donc nesting dans les niveaux inférieurs impossible).
Mais comme Benoit avait remonté plus haut, c'est peut être lié au fait que mes filtres ne sont pas assez permissifs finalement par rapport à ce qui avait été discuté dans les conventions.
Je refairai une passe la dessus